### PR TITLE
Partner Display and Reorder [Do Not Merge]

### PR DESCRIPTION
**Pull request by alvinwan**
Allows group members to switch and save order.

Front-end UI:

site-partner-switch

### CODE REVIEW

File: server/app/api.py [hunk 1]
```diff
@@ -1431,7 +1431,13 @@ class GroupAPI(APIResource):
             },
         'exit': {
             'methods': set(['PUT', 'POST']),
+            },
+        'save_order': {
```
Comment (soumyabasu) [inline server/app/api.py]: You should probably call this "reorder" instead of "save_order".

File: server/app/api.py [hunk 2]
```diff
@@ -1431,7 +1431,13 @@ class GroupAPI(APIResource):
...
+            'methods': {'PUT', 'POST'},
+            'web_args:': {
+                'order': Arg(dict, required=True)
```
Comment (soumyabasu) [inline server/app/api.py]: I was mentioning here-  you should take in a list of members here.

File: server/app/api.py [hunk 3]
```diff
@@ -1498,6 +1504,15 @@ def exit(self, group, user, data):
...
         group.exit(user)
+        
+    def save_order(self, group, user, data):
+        """ Saves order of partners """
+        need = Need('save_order')
+        if not group.can(user, need, group):
+            raise need.exception()
+
+        self.order = data['order']
+        self.save()
```
Comment (soumyabasu) [inline server/app/api.py]: Okay, so the logic here would go something like this:

new_order = data['order']

if len(new_order) != len(group.member):
    # Tell the frontend that they messed up

for member in group.member:
     if member not in new_order:
           # Tell the frontend they messed up again.

group.member = new_order

File: server/app/models.py [hunk 4]
```diff
@@ -1027,7 +1028,7 @@ def _can(cls, user, need, group, query):
             return False
         if action in ("get", "exit"):
             return user.key in group.member or user.key in group.invited
-        elif action in ("invite", "remove"):
+        elif action in ("invite", "remove", "save_order"):
```
Comment (soumyabasu) [inline server/app/models.py]: You'll also probably want to change this to "reorder"

**Comment by soumyabasu** [comment]
Okay, that's the part with save_order. Great work on the frontend stuff though! :)

**Comment by soumyabasu** [comment]
Is this ready to go?

File: server/app/api.py [hunk 5]
```diff
@@ -1757,6 +1763,28 @@ def exit(self, group, user, data):
...
 
         group.exit(user)
+        
+    def reorder(self, group, user, data):
+        """ Saves order of partners """
+        need = Need('reorder')
+        if not group.can(user, need, group):
+            raise need.exception()
+
+        print(data)
```
Comment (Sumukh) [inline server/app/api.py]: print statement should be removed.

File: server/app/templates/student.html [hunk 6]
```diff
@@ -17,7 +17,7 @@
         <script src="/static/vendor/jquery/dist/jquery.min.js"></script>
 
         <script src="/static/vendor/jquery-ui/jquery-ui.min.js"></script>
-        <script src="/static/vendor/raven-js/dist/raven.js"></script>
+        <script src="http://cdn.ravenjs.com/1.1.18/jquery,native/raven.min.js"></script>
```
Comment (Sumukh) [inline server/app/templates/student.html]: note: should be `//cdn.raven` (might be fixed in another PR)

**Comment by alvinwan** [comment]
@soumyabasu Not yet. I never figured it out; I'll continue working on it tonight or tomorrow.

**Comment by alvinwan** [comment]
Reordering now works.

partner-reorder

@Sumukh @soumyabasu Ready. ... I removed all print's and console.logs this time. :P

**Comment by soumyabasu** [comment]
The backend looks good to me.

@Sumukh - do you have any frontend comments? Otherwise, we're just waiting for the tests to pass.

File: server/app/templates/admin.html [hunk 7]
```diff
@@ -19,7 +19,7 @@
         <link href="/static/css/main.css" rel="stylesheet" type="text/css" />
 
         <link href="/static/vendor/angular-loading-bar/build/loading-bar.css" rel="stylesheet"></link>
-        <link href="/static/vendor/sweetalert/dist/sweetalert.css" rel="stylesheet"></link>
+        <link href="/static/vendor/sweetalert/dist/sweet-alert.css" rel="stylesheet"></link>
```
Comment (Sumukh) [inline server/app/templates/admin.html]: I think this should be sweetalert.css - I updated the bower file so you may need to run bower install too see that.

File: server/app/templates/admin.html [hunk 8]
```diff
@@ -45,7 +45,7 @@
         <script src="/static/vendor/angular-moment/angular-moment.js"></script>
         <script src="/static/vendor/showdown/compressed/Showdown.js"></script>
         <script src="/static/vendor/jquery.tablesorter/js/jquery.tablesorter.min.js"></script>
-        <script src="/static/vendor/sweetalert/dist/sweetalert.min.js"></script>
+        <script src="/static/vendor/sweetalert/dist/sweet-alert.min.js"></script>
```
Comment (Sumukh) [inline server/app/templates/admin.html]: ^  same idea.

File: server/app/templates/student.html [hunk 9]
```diff
@@ -32,7 +32,7 @@
         <script src="/static/js/other/moment-timezone-with-data.js"></script>
         <script src="/static/vendor/angular-moment/angular-moment.js"></script>
         <script src="/static/vendor/showdown/compressed/Showdown.js"></script>
-        <script src="/static/vendor/sweetalert/dist/sweetalert.min.js"></script>
+        <script src="/static/vendor/sweetalert/dist/sweet-alert.min.js"></script>
```
Comment (Sumukh) [inline server/app/templates/student.html]: same thing about sweetalert

**Comment by Sumukh** [comment]
Frontend looks good - but I think sweetalert has been updated on master to be a newer version that doesn't use a dash. You will have to run bower install to get the latest version 

Once you fix that and it works - we should be good to merge.
